vm/src/obj/objfunction.rs

```diff
@@ -313,10 +462,11 @@ impl PyFunction {
 
     #[cfg(feature = "jit")]
     #[pymethod(magic)]
-    fn jit(&self, vm: &VirtualMachine) -> PyResult<()> {
-        self.jitted_code
+    fn jit(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult<()> {
+        zelf.jitted_code
             .get_or_try_init(|| {
-                rustpython_jit::compile(&self.code.code)
+                let arg_types = PyFunction::get_jit_arg_types(&zelf, vm)?;
+                rustpython_jit::compile(&zelf.code.code, &arg_types)
```

Comment:
This is a lot of code in objfunction, maybe there could be a submodule `obj/objfunction/jitfunc.rs` or something to store all the logic for py<->jit conversion?